pyccel/plugins/Openmp/grammar/openmp.tx

```diff
@@ -0,0 +1,451 @@
+                   VERSION=OMP_4_5 omp_exprs=OmpTxScalarExpr
+             )')';
+
+untied_clause: name='untied' VERSION=OMP_4_5;
+
+mergeable_clause: name='mergeable' VERSION=OMP_4_5;
+
+depend_clause: name='depend' '('(
+                     VERSION=OMP_4_5 dependence_type=DependenceType ':' omp_exprs=OmpTxList
+             )')';
+
+ordered_depend_clause: name='depend' '('(
+                     VERSION=OMP_4_5 dependence_type='sink' ':' omp_exprs=OmpTxExpressionList
+                     | VERSION=OMP_4_5 dependence_type='source'
+             )')';
+
+
+priority_clause: name='priority' '('(
+                   VERSION=OMP_4_5 omp_exprs=OmpTxIntegerExpr
+             )')';
+
+grainsize_clause: name='grainsize' '('(
+                   VERSION=OMP_4_5 omp_exprs=OmpTxIntegerExpr
+             )')';
+
+num_tasks_clause: name='num_tasks' '('(
+                   VERSION=OMP_4_5 omp_exprs=OmpTxIntegerExpr
+             )')';
+
+device_clause: name='device' '('(
+                   VERSION=OMP_4_5 omp_exprs=OmpTxIntegerExpr
+             )')';
+
+map_clause: name='map' '('(
+                   VERSION=OMP_4_5 ((map_type_modifier=MapTypeModifier (',')?)? map_type=MapType ':')? omp_exprs=OmpTxList
+             )')';
+
+is_device_ptr_clause: name='is_device_ptr' '('(
+                   VERSION=OMP_4_5 omp_exprs=OmpTxList
+             )')';
+
+defaultmap_clause: name='defaultmap' '('(
+                   VERSION=OMP_4_5 implicit_behavior=DefaultMapImplicitBehavior ':' variable_category=DefaultMapVariableCategory
+             )')';
+
+num_teams_clause: name='num_teams' '('(
+                   VERSION=OMP_4_5 omp_exprs=OmpTxIntegerExpr
+             )')';
+
+thread_limit_clause: name='thread_limit' '('(
+                   VERSION=OMP_4_5 omp_exprs=OmpTxIntegerExpr
+             )')';
+
+dist_schedule_clause: name='dist_schedule' '('(
+                   VERSION=OMP_4_5 (kind='static')? omp_exprs=OmpTxIntegerExpr
+             )')';
+
+threads_clause: VERSION=OMP_4_5 name='threads';
+
+simd_clause: VERSION=OMP_4_5 name='simd';
+
+atomic_clause: VERSION=OMP_4_5 value=atomic_clause_values;
+
+memory_order_clause: VERSION=OMP_4_5 value='seq_cst';
+
+allocate_clause: name='allocate' '('(
+                   VERSION=OMP_5_0 omp_exprs=OmpTxList
+             )')';
+
+order_clause: name='order' '('
+                   VERSION=OMP_5_0 'concurrent'
+                   ')';
+
+////////////////////////////////////////////////////
+//              Clauses values
+////////////////////////////////////////////////////
+atomic_clause_values: 'read' | 'write' | 'update' | 'capture';
+OmpTxScalarExpr: value=InnerContent | ParenthesisContent;
+OmpTxIntegerExpr: value=InnerContent | ParenthesisContent;
+OmpTxConstantPositiveInteger: value=/[0-9]+/;
+OmpTxList: throw_away+=ID[','];
+ParenthesizedExpression: (('(' ParenthesizedExpression ')') | /[^()]*/)*; // TODO: find a better way to do this as this will not escape the parenthesis inside of a string
```

Comment:
Can you elaborate more on this TODO? Maybe I can help? Usually `.tx` files are designed to handle parsing in a way that is more explicit than regexes to avoid this kind of issue.

Reply:
Maybe you are just missing something that matches a string?